pycode/memilio-epidata/memilio/epidata_test/test_epidata_getCaseData.py

```diff
@@ -1012,17 +1060,78 @@ def test_rep_date(self, mocklcsv):
-            make_plot, moving_average, split_berlin, rep_date)
+            read_data=read_data, file_format=file_format,
+            out_folder=out_folder, no_raw=no_raw, start_date=start_date,
+            end_date=end_date, impute_dates=impute_dates, make_plot=make_plot,
+            moving_average=moving_average, split_berlin=split_berlin,
+            rep_date=rep_date)
 
         mocklcsv.assert_called()
         self.assertEqual(len(os.listdir(directory)), 27)
 
+    def test_get_case_data_timeframe(self):
+
+        read_data = True
+        file_format = 'json_timeasstring'
+        out_folder = self.path
+        no_raw = False
+        impute_dates = False
+        make_plot = False
+        moving_average = 0
+        split_berlin = False
+        rep_date = False
+
+        directory = os.path.join(out_folder, 'Germany/')
+        gd.check_dir(directory)
+
+        # write file
+        self.write_case_data(directory)
+        # check if expected file is written
+        self.assertEqual(len(os.listdir(directory)), 1)
+
+        # start and end date 
+        start_date=date(2020, 12, 24)
+        end_date=date(2021,5,17)
+
+        gcd.get_case_data(
+            read_data=read_data, file_format=file_format,
+            out_folder=out_folder, no_raw=no_raw, start_date=start_date,
+            end_date=end_date, impute_dates=impute_dates, make_plot=make_plot,
+            moving_average=moving_average, split_berlin=split_berlin,
+            rep_date=rep_date)
+
+        file = 'cases_all_germany.json'
+        f_read = os.path.join(directory, file)
+        df_germany_start_end_date = pd.read_json(f_read)
+
+        # get case data without start date and end date
+        gcd.get_case_data(
+            read_data=read_data, file_format=file_format,
+            out_folder=out_folder, no_raw=no_raw, impute_dates=impute_dates,
+            make_plot=make_plot, moving_average=moving_average,
+            split_berlin=split_berlin, rep_date=rep_date)
+
+        f_read = os.path.join(directory, file)
+        df_germany = pd.read_json(f_read)
+
+        # extract dates which are between start and end date
+        upperdate = datetime.strftime(end_date, '%Y-%m-%d')
+        lowerdate = datetime.strftime(start_date, '%Y-%m-%d')
+        df_germany = df_germany[df_germany[dd.EngEng['date']] <= upperdate]
+        df_germany = df_germany[df_germany[dd.EngEng['date']] >= lowerdate]
+
+        # dataframes should be equal
+        self.assertEqual(len(df_germany_start_end_date),len(df_germany),"Dataframes don't have the same length.")
+        self.assertEqual(list(df_germany['Confirmed']),list(df_germany['Confirmed']),"Dataframes don't have the same confirmed cases.")
+        self.assertEqual(list(df_germany['Recovered']),list(df_germany['Recovered']),"Dataframes don't have the same recovered cases.")
+        self.assertEqual(list(df_germany['Deaths']),list(df_germany['Deaths']),"Dataframes don't have the same death cases.")
```

Comment:
Don't you mean to compare df_germany_start_end_date against df_germany????

Reply:
Yes.